pkg/scale/encode.go

```diff
@@ -223,6 +225,35 @@ func (es *encodeState) encodeArray(in interface{}) (err error) {
 	return
 }
 
+func (es *encodeState) encodeMap(in interface{}) (err error) {
+	v := reflect.ValueOf(in)
+	err = es.encodeLength(v.Len())
+	if err != nil {
+		return
+	}
+
+	for i := v.MapRange(); i.Next(); {
```

Comment:
nit rename `i` to `keyValue`, since `i` is usually for an index which isn't the case here :thinking: